app/views/static/js/plugins.js

```diff
@@ -5,13 +5,19 @@ function _append(text) {
 }
 
 function _getTime(timestamp) {
-    if (typeof timestamp === "undefined")
-        return new Date().getHours() + ':' + new Date().getMinutes();
-    let currentDate = new Date((timestamp + new Date().getTimezoneOffset() * 60) * 1000);
+    let currentDate = undefined
+    if (typeof timestamp === 'string') {
+        currentDate = new Date(timestamp)
+        currentDate.setTime(currentDate.getTime() - new Date().getTimezoneOffset() * 60000)
+    } else if (typeof timestamp === "undefined") {
+        currentDate = new Date()
```

Comment:
```suggestion
    if (typeof timestamp === "string") {
        currentDate = new Date(timestamp)
        currentDate.setTime(currentDate.getTime() - new Date().getTimezoneOffset() * 60000)
    } else if (timestamp === null) {
        currentDate = new Date()
```
At least for my tests timestamp was never of type undefined,
the call to this function right upon sending a message passes an argument of type number e.g. 1623533243.557 this is correctly handled in the else case.
the call to this function upon refreshing a page when date_modified is null passes an argument of type object namely null, this was wrongly handled in the else case, resulting in a timestamp that was always 23:00
the call to this function upon refreshing a page when date_modified is not null passes an argument of type string e.g. 2021-06-12T19:46:27 this was correctly handled

I suggest this adaption because I think that is what one had in mind for this case but personally I would also suggest using the date_created attribute instead because now that date_modified is always null for messages that have not been changed (which will probably almost never happen) refreshing the page will always trigger the === null case, and this case then uses the current time, so after refreshing all messages will have the same inaccurate timestamp

Reply:
Nice observation! I applied your suggestion but also fixed the history plugins to pass the `date_created` if `date_modified` is null.

Reply:
While we are talking about the timestamp. Not sure whether your change has also fixed this problem. But when I send a message, I instantly get a correct timestamp shown (without having to refresh), but the other user (another tab) receives the same message but with a wrong timestamp (-2h). This was already the case in the old slurk.
The cause has to be before entering the _getTime function, as two different numbers (milliseconds passed after 1970 ...) are passed to the _getTime function. Maybe it has something to do with the way the timestamp for the message event is created? ``` 'timestamp': timegm(datetime.now().utctimetuple()) ```